### PR TITLE
Add Offline definitions

### LINKED ISSUE #5556: Definition request: Offline
Please add definitions for [Offline](https://github.com/HubSpot/offline).

### PR DESCRIPTION
**Pull request by cgwrench**
Fixes #5556.

### CODE REVIEW

File: offline-js/offline-js-tests.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// Test file for offline-js.
+/// <reference path="offline-js.d.ts" />
+
+Offline.options = {
+    checkOnLoad: false,
+    interceptRequests: true,
+	checks: {
```
Comment (DanielRosenwasser) [inline offline-js/offline-js-tests.ts]: There's a mix of tabs and spaces here

File: offline-js/offline-js.d.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    on(event: "confirmed-down", handler: (e: Event) => any, context?: any): void;
+    on(event: "checking", handler: (e: Event) => any, context?: any): void;
+    on(event: "reconnect:started", handler: (e: Event) => any, context?: any): void;
+    on(event: "reconnect:stopped", handler: (e: Event) => any, context?: any): void;
+    on(event: "reconnect:tick", handler: (e: Event) => any, context?: any): void;
+    on(event: "reconnect:connecting", handler: (e: Event) => any, context?: any): void;
+    on(event: "reconnect:failure", handler: (e: Event) => any, context?: any): void;
+    on(event: "requests:flush", handler: (e: Event) => any, context?: any): void;
+    on(event: "requests:hold", handler: (e: Event) => any, context?: any): void;
+    off(event: string, handler?: (e: Event) => any): void;
```
Comment (DanielRosenwasser) [inline offline-js/offline-js.d.ts]: You want this overload to be at the bottom.

File: offline-js/offline-js.d.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
+// Type definitions for Offline 0.7.14
+// Project: https://github.com/HubSpot/offline
+// Definitions by: Chris Wrench <https://github.com/cgwrench>
+// Definitions: https://github.com/borisyankov/DefinitelyTyped
+
+declare var Offline: {
+    options: OfflineOptions;
+    check: () => void;
+    state: string;    
+    on(event: string, handler: (e: Event) => any, context?: any): void;
```
Comment (DanielRosenwasser) [inline offline-js/offline-js.d.ts]: You want this overload to be at the bottom

File: offline-js/offline-js.d.ts (hunk 2 above)
Comment (DanielRosenwasser) [inline offline-js/offline-js.d.ts]: Add a newline above `off` so it is easier to distinguish the functions

**Comment by cgwrench** [comment]
@DanielRosenwasser Thanks for reviewing this. 3159d3f should fix these issues.

**Comment by vvakame** [comment]
*offline-js/offline-js.d.ts*

check list

* [ ] is collect [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)?
  * https://www.npmjs.com/package/offline-js
  * http://bower.io/search/?q=offline-js
  * others?
* [X] has a [test file](http://definitelytyped.org/guides/contributing.html#tests)? (offline-js/offline-js-tests.ts or others)
* [ ] pass the Travis-CI test?

**Comment by vvakame** [comment]
@DanielRosenwasser thanks!
@cgwrench thanks mate!

**Pull request merged**